projects/gateway-utilities/ignition/script-python/General/Perspective/Table/code.py

```diff
@@ -0,0 +1,94 @@
+"""
+General.Perspective.Table
+
+This script contains functions used with table components.
+"""
+
+def export_table_to_csv(table):
+	"""
+	DESCRIPTION: Exports the table to a CSV file
+	PARAMETERS:
+		table: Table object being exported
+	RETURNS: 
+	"""
+	table_dataset = get_dataset(table)
+	excel_data = system.dataset.toExcel(showHeaders=True, dataset=table_dataset)
+	system.perspective.download("data.xlsx", excel_data)
+
+def get_dataset(table):
+	"""
+	Description: Returns dataset built from table component
+	PARAMETERS:
+		table: Table component to get dataset from
+	RETURNS: dataset built from table component with proper formatting 
+	"""
+	table_data = []
+	render_list = [column['render'] for column in table.props.columns]
+	
+	# Add {"column0": value, "column1": value, ...} to represent the column titles
+	table_data.append({"column%s" % index: header_title for index, header_title in enumerate(get_column_titles(table))})
+	headers = ["column%s" % index for index in range(len(render_list))]
+	
+	for row in table.props.data:
+		temp_dict = {}
+		for index, val in enumerate(row.values()):
+			if render_list[index] == "date":
+				temp_dict.update({"column%s" % index: system.date.format(val, "yy/MM/dd HH:mm:ss")})
+			else:
+				temp_dict.update({"column%s" % index: round(val, 2)})
+		table_data.append(temp_dict)
+
+	return General.Conversion.convert_list_to_dataset(list_var=table_data, headers_list=headers)
+
+def set_column_attributes(field, title, datatype):
```

Comment:
I feel like the name of this is off, maybe something like a `get_column_template`?